src/sentry/static/sentry/app/components/errorRobot.tsx

```diff
@@ -1,79 +1,82 @@
 
 import {t} from 'app/locale';
 import Button from 'app/components/button';
 import CreateSampleEventButton from 'app/views/onboarding/createSampleEventButton';
 import withApi from 'app/utils/withApi';
-
-class ErrorRobot extends React.Component {
-  static propTypes = {
-    api: PropTypes.object,
-    org: PropTypes.object.isRequired,
-    project: PropTypes.object,
-
-    // sampleIssueId can have 3 values:
-    // - empty string to indicate it doesn't exist (render "create sample event")
-    // - non-empty string to indicate it exists (render "see sample event")
-    // - null/undefined to indicate the project API should be consulted to find out
-    sampleIssueId: PropTypes.string,
-
-    gradient: PropTypes.bool,
-  };
-
-  state = {
+import {Client} from 'app/api';
+import {LightWeightOrganization, Project} from 'app/types';
+
+type Props = {
+  api: Client;
+  org: LightWeightOrganization;
+  project?: Project;
+  gradient: boolean;
+  /**
+   * sampleIssueId can have 3 values:
+   * - empty string to indicate it doesn't exist (render "create sample event")
+   * - non-empty string to indicate it exists (render "see sample event")
+   * - undefined to indicate the project API should be consulted to find out
+   */
+  sampleIssueId?: string;
+};
+
+type State = {
+  error: boolean;
+  loading: boolean;
+  sampleIssueId?: string;
+};
+
+class ErrorRobot extends React.Component<Props, State> {
+  state: State = {
     error: false,
     loading: false,
     sampleIssueId: this.props.sampleIssueId,
   };
 
-  UNSAFE_componentWillMount() {
+  componentDidMount() {
     this.fetchData();
   }
 
-  fetchData() {
+  async fetchData() {
     const {org, project} = this.props;
     const {sampleIssueId} = this.state;
 
     if (!project) {
       return;
     }
 
-    if (sampleIssueId === null || sampleIssueId === undefined) {
-      const url = '/projects/' + org.slug + '/' + project.slug + '/issues/';
-      const requestParams = {limit: 1};
+    if (sampleIssueId !== null && sampleIssueId !== undefined) {
+      return;
+    }
+
+    const url = `/projects/${org.slug}/${project.slug}/issues/`;
+
+    this.setState({loading: true});
 
-      this.setState({loading: true});
-      this.props.api.request(url, {
+    try {
+      const data = await this.props.api.requestPromise(url, {
         method: 'GET',
-        data: requestParams,
-        success: data => {
-          this.setState({
-            loading: false,
-            sampleIssueId: (data.length > 0 && data[0].id) || '',
-          });
-        },
-        error: err => {
-          let error = err.responseJSON || true;
-          error = error.detail || true;
-          this.setState({
-            error,
-            loading: false,
-          });
-        },
+        data: {limit: 1},
       });
+      this.setState({sampleIssueId: (data.length > 0 && data[0].id) || ''});
+    } catch (err) {
+      let error = err.responseJSON || true;
+      error = error.detail || true;
```

Comment:
```suggestion
      const error = err.responseJSON?.detail || true;
```

Reply:
Let's go all the way and use `??` then too :)